ui/admin/tests/acceptance/credential-store/credentials/update-test.js

```diff
@@ -128,57 +218,118 @@ module(
-      await fillIn('[name="name"]', 'random string');
-      assert.strictEqual(currentURL(), urls.credential);
+      await fillIn('[name="name"]', MOCK_INPUT);
+      assert.strictEqual(currentURL(), urls.usernameKeyPairCredential);
       try {
         await visit(urls.credentials);
       } catch (e) {
-        assert.ok(find('.rose-dialog'));
-        await click('.rose-dialog-footer button:last-child', 'Click Cancel');
-        assert.strictEqual(currentURL(), urls.credential);
+        assert.dom('.rose-dialog').isVisible();
+        await click('.rose-dialog-footer button:first-child', 'Click Discard');
+        assert.strictEqual(currentURL(), urls.credentials);
         assert.notEqual(
-          this.server.schema.credentials.all().models[0].name,
-          'random string'
+          this.server.schema.credentials.where({ type: 'ssh_private_key' })
+            .models[0].name,
+          MOCK_INPUT
+        );
+      }
+    });
+
+    test('can cancel discard unsaved username & password credential changes via dialog', async function (assert) {
+      assert.expect(5);
+      const confirmService = this.owner.lookup('service:confirm');
+      confirmService.enabled = true;
+      assert.notEqual(instances.usernamePasswordCredential.name, MOCK_INPUT);
+      await visit(urls.usernamePasswordCredential);
+      await click('form [type="button"]', 'Activate edit mode');
+      const credentialName = find('[name="name"]').value;
+      await fillIn('[name="name"]', MOCK_INPUT);
+      assert.strictEqual(currentURL(), urls.usernamePasswordCredential);
+      try {
+        await visit(urls.credentials);
+      } catch (e) {
+        assert.dom('.rose-dialog').isVisible();
+        await click('.rose-dialog-footer button:last-child', 'Click Cancel');
+        assert.strictEqual(currentURL(), urls.usernamePasswordCredential);
+        assert.strictEqual(
+          this.server.schema.credentials.where({ type: 'username_password' })
+            .models[0].name,
+          credentialName
+        );
+      }
+    });
+
+    test('can cancel discard unsaved username & key pair credential changes via dialog', async function (assert) {
+      assert.expect(5);
+      const confirmService = this.owner.lookup('service:confirm');
+      confirmService.enabled = true;
+      assert.notEqual(instances.usernameKeyPairCredential.name, MOCK_INPUT);
+      await visit(urls.usernameKeyPairCredential);
+      await click('form [type="button"]', 'Activate edit mode');
+      const credentialName = find('[name="name"]').value;
+      await fillIn('[name="name"]', MOCK_INPUT);
+      assert.strictEqual(currentURL(), urls.usernameKeyPairCredential);
+      try {
+        await visit(urls.credentials);
+      } catch (e) {
+        assert.dom('.rose-dialog').isVisible();
+        await click('.rose-dialog-footer button:last-child', 'Click Cancel');
+        assert.strictEqual(currentURL(), urls.usernameKeyPairCredential);
```

Comment:
Maybe add one last assert to confirm that the `name` field still has the `MOCK_INPUT` value to show that cancel the discard didn't remove the form data

Reply:
I might also avoid ALL_CAPS constants for input only for consistency reasons.  The style has been to create regular vars inside each test case as needed for input.  As discussed, ALL_CAPS may become conventional of test selectors, but remains to be seen.  In any case, let's reserve ALL_CAPS for a shared convention.